### PR TITLE
Stackoverlow - more functionality

### PR DESCRIPTION
**Pull request by oladelmar**
Added search by questions, users and tags.
Search results are displayed but not styled.

I guess there is a more elegant way to do that. I just haven't found it yet)

### CODE REVIEW

File: src/components/Questions/Questions.vue [hunk 1]
```diff
@@ -26,7 +26,9 @@
           </v-flex>
           <v-flex>
             <div>
-              <router-link :to="{name: 'question', params: {id: question.id}}">{{ question.title }}</router-link>
+              <router-link :to="{name: 'question', params: {id: question.id}}">
```
Comment (gregurco) [inline src/components/Questions/Questions.vue]: Please check this rule: https://eslint.org/docs/rules/key-spacing

File: src/store/modules/results.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  state: {
+    resultQuestionsList: [],
+    resultUsersList: [],
+    resultTagsList: []
+  },
+  actions: {
+    fetchQuestions: ({ commit }, query) => {
+      fetchQuestionsSearchResults(query)
+        .then(result => {
+          console.log(result.data)
```
Comment (gregurco) [inline src/store/modules/results.js]: remove `console.log`

**Review by gregurco — CHANGES_REQUESTED** [review]

File: src/api/questions.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
 import API from './axios'
 
-export const fetchQuestions = (param) => API.get('/questions?_limit=10')
+// export const fetchQuestions = (param) => API.get('/questions?_limit=10')
+export const fetchQuestions = (params) => API.get(`/questions${params}`)
```
Comment (rapi) [inline src/api/questions.js]: Axios have params, dont send it via string

File: src/api/search.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import API from './axios'
+
+export const fetchQuestionsSearchResults = (param) => API.get('/questions?q='+ param)
+export const fetchUsersSearchResults = (param) => API.get('/users?q='+ param)
```
Comment (rapi) [inline src/api/search.js]: No string params

File: src/router/router.js [hunk 5]
```diff
@@ -33,6 +34,19 @@ export default new Router({
           props: true,
           component: () => import('@/views/Question.vue')
         },
+        {
+          path: '/search',
+          name: 'search',
+          params: ['query'],
+          props: (route) => ({ query: route.query.q }),
```
Comment (rapi) [inline src/router/router.js:41]: props: true

File: src/store/modules/questions.js [hunk 6]
```diff
@@ -7,11 +7,14 @@ export default {
...
+    fetch: ({ commit }, params) => {
       commit('mutateLoading', true)
-      fetchQuestions()
-        .then(result => commit('mutateQuestionList', result.data))
-        .then(() => commit('mutateLoading', false))
+      return fetchQuestions(params)
+        .then(result => {
+          commit('mutateQuestionList', result.data)
+          return result
+        })
```
Comment (rapi) [inline src/store/modules/questions.js:16]: make con catch on every request

File: src/store/modules/results.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import { fetchQuestionsSearchResults, fetchUsersSearchResults, fetchTagsSearchResults } from '@/api/search'
```
Comment (rapi) [inline src/store/modules/results.js]: you dont need this module, access the module(user, question) directly

File: src/store/modules/users.js [hunk 8]
```diff
@@ -7,11 +7,14 @@ export default {
...
+    fetch: ({ commit }, params) => {
       commit('mutateLoading', true)
-      fetchUsers()
-        .then(result => commit('mutateUserList', result.data))
-        .then(() => commit('mutateLoading', false))
+      return fetchUsers(params)
+        .then(result => {
+          commit('mutateUserList', result.data)
+          return result
+        })
```
Comment (rapi) [inline src/store/modules/users.js:17]: catch

File: src/views/AllSearchResults.vue [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+  name: 'AllSearchResults',
+  props: {
+    query: {
+      type: String,
+      required: true
+    }
+  },
+  computed: {
+    ...mapGetters({
+      resultQuestionsList: 'results/getResultQuestionsList',
```
Comment (rapi) [inline src/views/AllSearchResults.vue]: dont use general module, access the module directly

**Review by rapi — CHANGES_REQUESTED** [review]
Good job, just some logic changes

**Pull request merged**